x-pack/legacy/plugins/siem/public/pages/hosts/hosts_body.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { pure } from 'recompose';
+
+import { UseUrlState } from '../../components/url_state';
+import { GlobalTime } from '../../containers/global_time';
+
+import { indicesExistOrDataTemporarilyUnavailable, WithSource } from '../../containers/source';
+
+import { hostsModel, hostsSelectors, State } from '../../store';
+
+import { HostsComponentProps } from './hosts';
+import { scoreIntervalToDateTime } from '../../components/ml/score/score_interval_to_datetime';
+import { setAbsoluteRangeDatePicker as dispatchSetAbsoluteRangeDatePicker } from '../../store/inputs/actions';
+import { Anomaly } from '../../components/ml/types';
+
+const HostsBodyComponent = pure<HostsComponentProps>(
+  ({ filterQuery, setAbsoluteRangeDatePicker, children }) => {
+    return (
+      <WithSource sourceId="default">
+        {({ indicesExist, indexPattern }) =>
+          indicesExistOrDataTemporarilyUnavailable(indicesExist) ? (
+            <GlobalTime>
+              {({ to, from, setQuery }) => (
+                <UseUrlState indexPattern={indexPattern}>
+                  {({ isInitializing }) => (
+                    <>
+                      {typeof children === 'function' &&
```

Comment:
I think it was because I didn’t declare the type correctly that's why it's confused and complained about null or undefined. Just add it to the type declaration and remove the check here.